app/lib/use-effect-no-strict-mode.ts

```diff
@@ -16,22 +16,18 @@ import {
 export const useEffectNoStrictMode = (
   effect: EffectCallback,
   deps?: DependencyList,
-  name?: string,
 ) => {
   const hasRunOnce = useRef(false);
 
   // biome-ignore lint/correctness/useExhaustiveDependencies: we are ignoring this because biome sees effect as missing dep. We can safely do that because effect will change only when the deps change too.
   useEffect(() => {
     // Strict mode is on only in development mode so no need to have special handling if we are not running in that mode
-    const isDevelopmentMode = import.meta.env.NODE_ENV === 'development';
+    const nodeEnv: string = import.meta.env.NODE_ENV ?? 'development';
```

Comment:
there was a bug there where NODE_ENV was undefined when running `bun run dev`. We want to treat undefined as `development`